packages/solana-v1-contrib/src/retry_tx_sender.ts

```diff
@@ -182,6 +211,56 @@ export class RetryTxSender {
     }
   }
 
+  /**
+   * Fetch a transaction sent by this sender by polling getTransaction.
+   *
+   * Stops polling once
+   *   1. The transaction is fetched
+   *   2. The transaction times out (via timeout promise or lastValidBlockHeight)
+   *   3. Confirmation is cancelled via cancelConfirm
+   *
+   * Notes:
+   *   * After confirming, subsequent calls will return a cached tx
+   *   * tryFetchTx should not be invoked multiple times in parallel
+   *   * tryFetchTx should not be invoked in parallel with tryConfirm
+   *
+   * @param lastValidBlockHeight cancel tx confirmation loop once this block height is reached
+   * @param opts {
+   *   @param disableWs don't listen for onSignature WS events when confirming
+   * }
+   */
+  async tryFetchTx(
+    lastValidBlockHeight?: number,
+  ): Promise<VersionedTransactionResponse> {
+    if (this.fetchedTx) {
+      this.logger?.info('✅ Tx already fetched');
+      return this.fetchedTx;
+    }
+
+    if (!this.txSig) {
+      throw new Error('you need to send the tx first');
+    }
+
+    this.done = false;
+    try {
+      this.fetchedTx = await this._fetchTransaction(
+        this.txSig,
+        lastValidBlockHeight,
+      );
+      this.confirmedTx = {
+        txSig: this.txSig,
+        slot: this.fetchedTx.slot,
+        err: this.fetchedTx.meta?.err ?? null,
+      };
+      return this.fetchedTx;
+    } catch (e) {
+      this.logger?.error(`${JSON.stringify(e)}`);
+      throw e;
+    } finally {
+      this._stopWaiting();
+    }
+  }
```

Comment:
Yup this wouldn't have WS, just polling. Haven't had the chance to look into WS in tryConfirm yet, will do tn.

Reply:
After some testing, it looks like WS confirming is reliable. Both WS and polling at 300 ms intervals are roughly equally likely to pick up a confirmed status, and turning the poll interval up from 2000 ms wasn't strictly necessary since WS would have picked it up.

Polling getTransaction at 300ms intervals appears to yield similar results to WS + 300 ms getSignatureStatus poll, with the added benefit of eliminating the additional call to fetch the actual tx. So the getTransaction approach is promising.